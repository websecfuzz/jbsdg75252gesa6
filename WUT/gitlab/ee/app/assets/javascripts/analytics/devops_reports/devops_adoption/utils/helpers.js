import { sprintf } from '~/locale';
import { joinPaths } from '~/lib/utils/url_utility';
import { GROUP_DEVOPS_PATH } from '../constants';

/**
 * A helper function which accepts the enabledNamespaces,
 *
 * @param {Object} params the enabledNamespaces data, and the check for open modals
 *
 * @return {Boolean} a boolean to determine if table data should be polled
 */
export const shouldPollTableData = ({ enabledNamespaces, openModal }) => {
  if (openModal) {
    return false;
  }
  if (!enabledNamespaces.length) {
    return true;
  }

  const anyPendingEnabledNamespaces = enabledNamespaces.some(
    (node) => node.latestSnapshot === null,
  );

  return anyPendingEnabledNamespaces;
};

/**
 * A helper function which extracts the total feature adoption count for a group
 * of snapshot data, filtered out by specific features / columns
 *
 * @param { Array } snapshots the snapshot data for a given group node
 * @param { Array } cols the columns which need to be used for the calculation
 *
 * @return { Array } an array containing the adopted counts for the given columns
 */
export const getAdoptedCountsByCols = (snapshots, cols) => {
  return snapshots.reduce((acc, snapshot) => {
    const adoptedCount = cols.reduce((adopted, col) => {
      return snapshot[col.key] ? adopted + 1 : adopted;
    }, 0);

    return [...acc, adoptedCount];
  }, []);
};

/**
 * A helper function which computes the DevOps Adoption feature path
 * given a specific group path
 *
 * @param { String } fullPath the full path for the group
 *
 * @return { String } the path for the group level DevOps Adoption feature
 */
export const getGroupAdoptionPath = (fullPath) => {
  const prefix = gon.relative_url_root || '';
  return fullPath ? joinPaths(prefix, sprintf(GROUP_DEVOPS_PATH, { fullPath })) : null;
};
