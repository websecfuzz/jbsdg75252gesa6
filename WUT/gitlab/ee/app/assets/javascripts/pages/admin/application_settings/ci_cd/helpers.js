/**
 * Filters [items] based on a given [searchTerm].
 * Catagories with no items after filtering are not included in the returned object.
 * @param {Object} allItems - { <categoryName>: [{ name, id }] }
 * @param {String} searchTerm
 * @returns {Object}
 */
export function filterItems(allItems, searchTerm) {
  return Object.entries(allItems)
    .map(([key, items]) => ({
      text: key,
      options: items
        .filter((item) => item.name.toLowerCase().includes(searchTerm))
        .map((item) => ({
          text: item.name,
          value: item.key,
        })),
    }))
    .filter((group) => group.options.length > 0);
}
