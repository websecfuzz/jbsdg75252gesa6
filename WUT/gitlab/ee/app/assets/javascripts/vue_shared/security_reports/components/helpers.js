import { __ } from '~/locale';
import { EMPTY_BODY_MESSAGE } from './constants';

/**
 * A helper function which validates the passed
 * in body string.
 *
 * It returns an empty string if the body has explicitly
 * been passed in as an empty string, a fallback
 * message if the body is null / undefined, else
 * it will return the original body string.
 *
 * @param {String} body the body message
 *
 * @return {String} the validated body message
 */
export const bodyWithFallBack = (body) => (body === '' ? '' : body || EMPTY_BODY_MESSAGE);

/**
 * Returns a string representation of the HTTP request
 *
 * @param {Object} httpData
 * @param {String} httpData.body
 * @param {String} httpData.method
 * @param {String} httpData.url
 * @param {{name: String, value: String}[]} httpData.headers
 *
 * @returns {String}
 */
export const getHttpString = (httpData) => {
  if (!httpData) {
    return '';
  }

  const { body, headers, method, url, statusCode, reasonPhrase } = httpData;
  const methodOrStatusCode = method || statusCode;
  const urlOrReasonPhrase = url || reasonPhrase;
  const headerString = headers.map(({ name, value }) => `${name}: ${value}`).join('\n');

  return `${methodOrStatusCode} ${urlOrReasonPhrase}\n${headerString}\n\n${bodyWithFallBack(body)}`;
};

export const getDismissalNoteEventText = ({ hasProject, hasPipeline, hasDismissalReason }) => {
  if (hasDismissalReason) {
    if (hasPipeline && hasProject) {
      return __(
        '%{statusStart}Dismissed%{statusEnd}: %{dismissalReason} on pipeline %{pipelineLink} at %{projectLink}',
      );
    }

    if (hasPipeline) {
      return __(
        '%{statusStart}Dismissed%{statusEnd}: %{dismissalReason} on pipeline %{pipelineLink}',
      );
    }

    if (hasProject) {
      return __('%{statusStart}Dismissed%{statusEnd}: %{dismissalReason} at %{projectLink}');
    }

    return __('%{statusStart}Dismissed%{statusEnd}: %{dismissalReason}');
  }

  if (hasPipeline && hasProject) {
    return __('%{statusStart}Dismissed%{statusEnd} on pipeline %{pipelineLink} at %{projectLink}');
  }

  if (hasPipeline) {
    return __('%{statusStart}Dismissed%{statusEnd} on pipeline %{pipelineLink}');
  }

  if (hasProject) {
    return __('%{statusStart}Dismissed%{statusEnd} at %{projectLink}');
  }

  return __('%{statusStart}Dismissed%{statusEnd}');
};
