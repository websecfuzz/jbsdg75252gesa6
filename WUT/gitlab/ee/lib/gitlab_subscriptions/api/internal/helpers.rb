# frozen_string_literal: true

module GitlabSubscriptions
  module API
    module Internal
      module Helpers
        def jwt_request?
          headers[Auth::INTERNAL_API_REQUEST_HEADER].present?
        end

        def authenticate_from_jwt!
          unauthorized! unless Auth.verify_api_request(headers)
        end
      end
    end
  end
end
