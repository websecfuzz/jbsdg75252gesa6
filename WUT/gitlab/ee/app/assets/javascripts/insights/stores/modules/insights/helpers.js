import { pairDataAndLabels, buildNullSeries } from 'ee/analytics/shared/utils';
import { BASE_NULL_SERIES_OPTIONS, BASE_SERIES_DATA_OPTIONS } from 'ee/analytics/shared/constants';
import {
  CHART_TYPES,
  INSIGHTS_DATA_SOURCE_DORA,
  INSIGHTS_NO_DATA_TOOLTIP,
} from 'ee/insights/constants';
import { __, s__ } from '~/locale';

const getGroupByValue = (queryParams) => {
  return queryParams.group_by || queryParams.params.group_by;
};

const getTypeValue = (queryParams) => {
  return queryParams.issuable_type || queryParams.params.issuable_type || queryParams.params.metric;
};

const getAxisTitle = (label) => {
  switch (label) {
    case 'day':
      return __('Days');
    case 'week':
      return __('Weeks');
    case 'month':
      return __('Months');
    case 'issue':
      return __('Issues');
    case 'merge_request':
      return __('Merge requests');
    case 'deployment_frequency':
      return s__('DORA4Metrics|Deployment frequency');
    case 'lead_time_for_changes':
      return s__('DORA4Metrics|Lead time for changes (median days)');
    case 'time_to_restore_service':
      return s__('DORA4Metrics|Time to restore service (median days)');
    case 'change_failure_rate':
      return s__('DORA4Metrics|Change failure rate (percentage)');
    default:
      return '';
  }
};

export const transformChartDataForGlCharts = (
  { type, query: queryParams },
  { labels, datasets },
) => {
  const yAxisTitle = getAxisTitle(getTypeValue(queryParams));
  const formattedData = {
    xAxisTitle: getAxisTitle(getGroupByValue(queryParams)),
    yAxisTitle,
    labels,
    datasets: [],
    seriesNames: [],
  };

  const { data_source: dataSource = '' } = queryParams;

  switch (type) {
    case CHART_TYPES.BAR:
      formattedData.datasets = [
        {
          name: 'all',
          data: labels.map((label, i) => [label, datasets[0].data[i]]),
        },
      ];
      break;
    case CHART_TYPES.STACKED_BAR:
      formattedData.datasets.push(
        ...datasets.map((dataset) => ({
          name: dataset.label,
          data: dataset.data,
        })),
      );
      break;
    case CHART_TYPES.LINE:
      if (dataSource === INSIGHTS_DATA_SOURCE_DORA) {
        const paired = pairDataAndLabels({
          datasetNames: [yAxisTitle],
          axisLabels: labels,
          datasets,
        });

        formattedData.datasets = buildNullSeries({
          seriesData: paired,
          seriesDataOptions: BASE_SERIES_DATA_OPTIONS,
          nullSeriesTitle: INSIGHTS_NO_DATA_TOOLTIP,
          nullSeriesOptions: BASE_NULL_SERIES_OPTIONS,
        });
      } else {
        formattedData.datasets.push(
          ...datasets.map((dataset) => ({
            name: dataset.label,
            data: labels.map((label, i) => [label, dataset.data[i]]),
          })),
        );
      }
      break;
    default:
      formattedData.datasets = { all: labels.map((label, i) => [label, datasets[0].data[i]]) };
  }
  return formattedData;
};

export default {
  transformChartDataForGlCharts,
};
