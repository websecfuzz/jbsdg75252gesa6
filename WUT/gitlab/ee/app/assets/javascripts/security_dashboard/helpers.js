import isPlainObject from 'lodash/isPlainObject';
import { REPORT_TYPES_ALL } from 'ee/security_dashboard/constants';
import { convertObjectPropsToSnakeCase } from '~/lib/utils/common_utils';
import { __ } from '~/locale';

/**
 * Provided a security reports summary from the GraphQL API, this returns an array of arrays
 * representing a properly formatted report ready to be displayed in the UI. Each sub-array consists
 * of the user-friend report's name, and the summary's payload. Note that summary entries are
 * considered empty and are filtered out of the return if the payload is `null` or don't include
 * a vulnerabilitiesCount property. Report types whose name can't be matched to a user-friendly
 * name are filtered out as well.
 *
 * Take the following summary for example:
 * {
 *   containerScanning: { vulnerabilitiesCount: 123 },
 *   invalidReportType: { vulnerabilitiesCount: 123 },
 *   dast: null,
 * }
 *
 * The formatted summary would look like this:
 * [
 *   ['containerScanning', { vulnerabilitiesCount: 123 }]
 * ]
 *
 * Note that `invalidReportType` was filtered out as it can't be matched with a user-friendly name,
 * and the DAST report was omitted because it's empty (`null`).
 *
 * @param {Object} rawSummary
 * @returns {Array}
 */
export const getFormattedSummary = (rawSummary = {}) => {
  if (!isPlainObject(rawSummary)) {
    return [];
  }
  // Convert keys to snake case so they can be matched against REPORT_TYPES keys for translation
  const snakeCasedSummary = convertObjectPropsToSnakeCase(rawSummary);
  // Convert object to an array of entries to make it easier to loop through
  const summaryEntries = Object.entries(snakeCasedSummary);
  // Filter out empty entries as we don't want to display those in the summary
  const withoutEmptyEntries = summaryEntries.filter(
    ([, scanSummary]) => scanSummary?.vulnerabilitiesCount !== undefined,
  );
  // Replace keys with translations found in REPORT_TYPES if available
  const formattedEntries = withoutEmptyEntries.map(([scanType, scanSummary]) => {
    const name = REPORT_TYPES_ALL[scanType];
    return name ? [name, scanSummary] : null;
  });
  // Filter out keys that could not be matched with any translation and are thus considered invalid
  return formattedEntries.filter((entry) => entry !== null);
};

/**
 * Limits the number of projects displayed per vulnerability grade.
 *
 * Takes an array of vulnerability grade data and ensures that each grade
 * shows at most the specified number of projects. Grades with fewer projects
 * than the limit are returned unchanged.
 *
 * @param {Array} vulnerabilityGradesQueryResults - Array of vulnerability grade objects from GraphQL query
 * @param {number} [maxProjects=5] - Maximum number of projects to show per grade
 * @returns {Array} Array of vulnerability grade objects with limited project nodes
 */
export const limitVulnerabilityGradeProjects = (
  vulnerabilityGradesQueryResults,
  maxProjects = 5,
) => {
  return vulnerabilityGradesQueryResults.map((gradeData) => {
    const nodes = gradeData.projects?.nodes;

    if (!nodes || nodes.length <= maxProjects) {
      return gradeData;
    }

    return {
      ...gradeData,
      projects: {
        ...gradeData.projects,
        nodes: nodes.slice(0, maxProjects),
      },
    };
  });
};

export const PROJECT_LOADING_ERROR_MESSAGE = __('An error occurred while retrieving projects.');

/**
 * Custom error class for PDF export operations that are not ready.
 *
 * @param {string} message - User-friendly error message describing why the export isn't ready
 */
export class PdfExportError extends Error {
  constructor(message) {
    super(message);
    this.name = 'PdfExportError';
  }
}

export default () => ({});
