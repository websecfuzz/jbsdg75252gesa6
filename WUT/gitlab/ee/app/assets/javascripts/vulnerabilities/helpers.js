import { getIdFromGraphQLId } from '~/graphql_shared/utils';
import { isAbsolute, isValidURL } from '~/lib/utils/url_utility';
import {
  REGEXES,
  SUPPORTED_IDENTIFIER_TYPE_CWE,
  SUPPORTED_IDENTIFIER_TYPE_OWASP,
  VULNERABILITY_TAB_INDEX_TO_NAME,
} from './constants';

// Get the issue in the format expected by the descendant components of related_issues_block.vue.
export const getFormattedIssue = (issue) => ({
  ...issue,
  reference: `#${issue.iid}`,
  path: issue.web_url,
});

export const getAddRelatedIssueRequestParams = (reference, defaultProjectId) => {
  let issueId = reference;
  let projectId = defaultProjectId;

  // If the reference is an issue number, parse out just the issue number.
  if (REGEXES.ISSUE_FORMAT.test(reference)) {
    [, issueId] = REGEXES.ISSUE_FORMAT.exec(reference);
  }
  // If the reference is an absolute URL and matches the issues URL format, parse out the project and issue.
  else if (isValidURL(reference) && isAbsolute(reference)) {
    const { pathname } = new URL(reference);

    if (REGEXES.LINK_FORMAT.test(pathname)) {
      [, projectId, issueId] = REGEXES.LINK_FORMAT.exec(pathname);
    }
  }

  return { target_issue_iid: issueId, target_project_id: projectId };
};

export const normalizeGraphQLNote = (note) => {
  if (!note) {
    return null;
  }

  return {
    ...note,
    id: getIdFromGraphQLId(note.id),
    author: {
      ...note.author,
      id: getIdFromGraphQLId(note.author.id),
      path: note.author.webPath,
    },
  };
};

export const normalizeGraphQLVulnerability = (vulnerability) => {
  if (!vulnerability) {
    return null;
  }

  const newVulnerability = { ...vulnerability };

  if (vulnerability.id) {
    newVulnerability.id = getIdFromGraphQLId(vulnerability.id);
  }

  if (vulnerability.state) {
    newVulnerability.state = vulnerability.state.toLowerCase();
  }

  ['confirmed', 'resolved', 'dismissed'].forEach((state) => {
    if (vulnerability[`${state}By`]?.id) {
      newVulnerability[`${state}ById`] = getIdFromGraphQLId(vulnerability[`${state}By`].id);
      delete newVulnerability[`${state}By`];
    }
  });

  return newVulnerability;
};

export const normalizeGraphQLLastStateTransition = (graphQLVulnerability, vulnerability) => {
  const stateTransitions = [...vulnerability.stateTransitions];

  // The vulnerability status mutation only returns 1 stateTransition
  const [graphQLLastStateTransitions] = graphQLVulnerability.stateTransitions.nodes;
  stateTransitions.push({
    ...graphQLLastStateTransitions,
    dismissalReason: graphQLLastStateTransitions.dismissalReason?.toLowerCase(),
  });

  return { stateTransitions };
};

export const formatIdentifierExternalIds = ({ externalType, externalId, name }) => {
  return `[${externalType}]-[${externalId}]-[${name}]`;
};

export const isSupportedIdentifier = (externalType) => {
  return (
    externalType?.toLowerCase() === SUPPORTED_IDENTIFIER_TYPE_CWE ||
    // Case matters here. owasp and OWASP require different configuration
    // Currently, our API only support lowercase owasp
    // Uppercase OWASP will be supported in a follow up issue:
    // https://gitlab.com/gitlab-org/gitlab/-/issues/366556
    externalType === SUPPORTED_IDENTIFIER_TYPE_OWASP
  );
};

export const getRefFromBlobPath = (path) => {
  // Matches the 40-character hex string of the Git SHA in the blob path, for example:
  // "/group/project/-/blob/cdeda7ae724a332e008d17245209d5edd9ba6499/src/file.js"
  // will match "cdeda7ae724a332e008d17245209d5edd9ba6499".
  if (typeof path === 'string') {
    const match = path.match(/blob\/([a-f0-9]{40})/);
    return match ? match[1] : '';
  }
  return '';
};

/**
 *  Determines the tab index for code flow based on the route query parameters.
 */
export const getTabIndexForCodeFlowPage = (route) => {
  if (route?.query?.tab) {
    return 1;
  }
  return 0;
};

/**
 * @typedef {Object} TabNavigationParams
 * @property {string} path - The current route path.
 * @property {number} index - The index of the tab to be set.
 * @property {number} [selectedIndex] - The currently selected tab index. This is optional as it might be undefined.
 */

/**
 * Updates the route to reflect the selected tab index.
 * This function is used to update the URL when switching between different views
 * of the vulnerability, between 'details' and 'code flow'
 * @param {Object} router - The router object used for navigation.
 * @param {TabNavigationParams} routeProps - The route properties including path and tab indices.
 */
export const setTabIndexForCodeFlowPage = (router, routeProps) => {
  const { path, index, selectedIndex } = routeProps;

  // Don't set the tab index if it's the same. This prevents some querystring bugs.
  if (index === selectedIndex) {
    return;
  }
  router.push({
    path,
    query: { tab: VULNERABILITY_TAB_INDEX_TO_NAME[index] },
  });
};
