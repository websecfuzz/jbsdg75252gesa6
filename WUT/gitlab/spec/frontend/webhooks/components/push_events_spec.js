import { nextTick } from 'vue';
import { GlFormCheckbox, GlFormRadioGroup } from '@gitlab/ui';
import { shallowMountExtended } from 'helpers/vue_test_utils_helper';
import PushEvents from '~/webhooks/components/push_events.vue';

describe('Webhook push events form editor component', () => {
  let wrapper;

  const findPushEventsCheckBox = (w = wrapper) => w.findComponent(GlFormCheckbox);
  const findPushEventsIndicator = (w = wrapper) => w.find('input[name="hook[push_events]"]');
  const findPushEventRulesGroup = (w = wrapper) => w.findComponent(GlFormRadioGroup);
  const getPushEventsRuleValue = (w = wrapper) => findPushEventRulesGroup(w).vm.$attrs.checked;
  const findWildcardRuleInput = (w = wrapper) => w.findByTestId('webhook_branch_filter_field');
  const findRegexRuleInput = (w = wrapper) => w.findByTestId('webhook_branch_filter_field');

  const createComponent = (provides) =>
    shallowMountExtended(PushEvents, {
      provide: {
        isNewHook: true,
        pushEvents: false,
        strategy: 'wildcard',
        pushEventsBranchFilter: '',
        ...provides,
      },
    });

  describe('Renders push events checkbox', () => {
    it('when it is a new hook', async () => {
      wrapper = createComponent({
        isNewHook: true,
      });
      await nextTick();

      const checkbox = findPushEventsCheckBox();
      expect(checkbox.exists()).toBe(true);
      expect(findPushEventRulesGroup().exists()).toBe(false);
      expect(findPushEventsIndicator().attributes('value')).toBe('false');
    });

    it('when it is not a new hook and push events is enabled', async () => {
      wrapper = createComponent({
        isNewHook: false,
        pushEvents: true,
      });
      await nextTick();

      expect(findPushEventsCheckBox().exists()).toBe(true);
      expect(findPushEventRulesGroup().exists()).toBe(true);
      expect(findPushEventsIndicator().attributes('value')).toBe('true');
    });
  });

  describe('Different push events rules', () => {
    describe('when editing new hook', () => {
      beforeEach(async () => {
        wrapper = createComponent({
          isNewHook: true,
        });
        await nextTick();
        await findPushEventsCheckBox().vm.$emit('input', true);
        await nextTick();
      });

      it('all_branches should be selected by default', () => {
        expect(findPushEventRulesGroup().element).toMatchSnapshot();
      });

      it('should be able to set wildcard rule', async () => {
        expect(getPushEventsRuleValue()).toBe('all_branches');
        expect(findWildcardRuleInput().exists()).toBe(false);
        expect(findRegexRuleInput().exists()).toBe(false);

        await findPushEventRulesGroup(wrapper).vm.$emit('input', 'wildcard');
        expect(findWildcardRuleInput().exists()).toBe(true);
        expect(findPushEventRulesGroup().element).toMatchSnapshot();

        const testVal = 'test-val';
        findWildcardRuleInput().vm.$emit('input', testVal);
        await nextTick();
        expect(findWildcardRuleInput().attributes('value')).toBe(testVal);
      });

      it('should be able to set regex rule', async () => {
        expect(getPushEventsRuleValue()).toBe('all_branches');
        expect(findRegexRuleInput().exists()).toBe(false);
        expect(findWildcardRuleInput().exists()).toBe(false);

        await findPushEventRulesGroup(wrapper).vm.$emit('input', 'regex');
        expect(findRegexRuleInput().exists()).toBe(true);
        expect(findPushEventRulesGroup().element).toMatchSnapshot();

        const testVal = 'test-val';
        findRegexRuleInput().vm.$emit('input', testVal);
        await nextTick();
        expect(findRegexRuleInput().attributes('value')).toBe(testVal);
      });
    });

    describe('when editing existing hook', () => {
      it.each(['all_branches', 'wildcard', 'regex'])(
        'with "%s" strategy selected',
        async (strategy) => {
          wrapper = createComponent({
            isNewHook: false,
            pushEvents: true,
            pushEventsBranchFilter: 'foo',
            strategy,
          });
          await nextTick();

          expect(findPushEventsIndicator().attributes('value')).toBe('true');
          expect(findPushEventRulesGroup().element).toMatchSnapshot();
        },
      );
    });
  });
});
