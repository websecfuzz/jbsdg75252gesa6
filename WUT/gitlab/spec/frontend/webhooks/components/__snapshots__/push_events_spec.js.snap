// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webhook push events form editor component Different push events rules when editing existing hook with "all_branches" strategy selected 1`] = `
<gl-form-radio-group-stub
  checked="all_branches"
  disabledfield="disabled"
  htmlfield="html"
  name="hook[branch_filter_strategy]"
  options=""
  textfield="text"
  valuefield="value"
>
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_all_branches"
    value="all_branches"
  >
    <div>
      All branches
    </div>
  </gl-form-radio-stub>
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_wildcard"
    value="wildcard"
  >
    <div>
      Wildcard pattern
    </div>
  </gl-form-radio-stub>
  <div
    class="gl-ml-6"
  />
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_regex"
    value="regex"
  >
    <div>
      Regular expression
    </div>
  </gl-form-radio-stub>
  <div
    class="gl-ml-6"
  />
</gl-form-radio-group-stub>
`;

exports[`Webhook push events form editor component Different push events rules when editing existing hook with "regex" strategy selected 1`] = `
<gl-form-radio-group-stub
  checked="regex"
  disabledfield="disabled"
  htmlfield="html"
  name="hook[branch_filter_strategy]"
  options=""
  textfield="text"
  valuefield="value"
>
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_all_branches"
    value="all_branches"
  >
    <div>
      All branches
    </div>
  </gl-form-radio-stub>
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_wildcard"
    value="wildcard"
  >
    <div>
      Wildcard pattern
    </div>
  </gl-form-radio-stub>
  <div
    class="gl-ml-6"
  />
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_regex"
    value="regex"
  >
    <div>
      Regular expression
    </div>
  </gl-form-radio-stub>
  <div
    class="gl-ml-6"
  >
    <gl-form-input-stub
      data-testid="webhook_branch_filter_field"
      name="hook[push_events_branch_filter]"
      type="text"
      value="foo"
    />
  </div>
  <p
    class="custom-control form-text gl-text-subtle"
  >
    <gl-sprintf-stub
      message="Regular expressions such as %{REGEX_CODE} are supported."
    />
  </p>
</gl-form-radio-group-stub>
`;

exports[`Webhook push events form editor component Different push events rules when editing existing hook with "wildcard" strategy selected 1`] = `
<gl-form-radio-group-stub
  checked="wildcard"
  disabledfield="disabled"
  htmlfield="html"
  name="hook[branch_filter_strategy]"
  options=""
  textfield="text"
  valuefield="value"
>
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_all_branches"
    value="all_branches"
  >
    <div>
      All branches
    </div>
  </gl-form-radio-stub>
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_wildcard"
    value="wildcard"
  >
    <div>
      Wildcard pattern
    </div>
  </gl-form-radio-stub>
  <div
    class="gl-ml-6"
  >
    <gl-form-input-stub
      data-testid="webhook_branch_filter_field"
      name="hook[push_events_branch_filter]"
      type="text"
      value="foo"
    />
  </div>
  <p
    class="custom-control form-text gl-text-subtle"
  >
    <gl-sprintf-stub
      message="Wildcards such as %{WILDCARD_CODE_STABLE} or %{WILDCARD_CODE_PRODUCTION} are supported."
    />
  </p>
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_regex"
    value="regex"
  >
    <div>
      Regular expression
    </div>
  </gl-form-radio-stub>
  <div
    class="gl-ml-6"
  />
</gl-form-radio-group-stub>
`;

exports[`Webhook push events form editor component Different push events rules when editing new hook all_branches should be selected by default 1`] = `
<gl-form-radio-group-stub
  checked="all_branches"
  disabledfield="disabled"
  htmlfield="html"
  name="hook[branch_filter_strategy]"
  options=""
  textfield="text"
  valuefield="value"
>
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_all_branches"
    value="all_branches"
  >
    <div>
      All branches
    </div>
  </gl-form-radio-stub>
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_wildcard"
    value="wildcard"
  >
    <div>
      Wildcard pattern
    </div>
  </gl-form-radio-stub>
  <div
    class="gl-ml-6"
  />
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_regex"
    value="regex"
  >
    <div>
      Regular expression
    </div>
  </gl-form-radio-stub>
  <div
    class="gl-ml-6"
  />
</gl-form-radio-group-stub>
`;

exports[`Webhook push events form editor component Different push events rules when editing new hook should be able to set regex rule 1`] = `
<gl-form-radio-group-stub
  checked="regex"
  disabledfield="disabled"
  htmlfield="html"
  name="hook[branch_filter_strategy]"
  options=""
  textfield="text"
  valuefield="value"
>
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_all_branches"
    value="all_branches"
  >
    <div>
      All branches
    </div>
  </gl-form-radio-stub>
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_wildcard"
    value="wildcard"
  >
    <div>
      Wildcard pattern
    </div>
  </gl-form-radio-stub>
  <div
    class="gl-ml-6"
  />
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_regex"
    value="regex"
  >
    <div>
      Regular expression
    </div>
  </gl-form-radio-stub>
  <div
    class="gl-ml-6"
  >
    <gl-form-input-stub
      data-testid="webhook_branch_filter_field"
      name="hook[push_events_branch_filter]"
      type="text"
      value=""
    />
  </div>
  <p
    class="custom-control form-text gl-text-subtle"
  >
    <gl-sprintf-stub
      message="Regular expressions such as %{REGEX_CODE} are supported."
    />
  </p>
</gl-form-radio-group-stub>
`;

exports[`Webhook push events form editor component Different push events rules when editing new hook should be able to set wildcard rule 1`] = `
<gl-form-radio-group-stub
  checked="wildcard"
  disabledfield="disabled"
  htmlfield="html"
  name="hook[branch_filter_strategy]"
  options=""
  textfield="text"
  valuefield="value"
>
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_all_branches"
    value="all_branches"
  >
    <div>
      All branches
    </div>
  </gl-form-radio-stub>
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_wildcard"
    value="wildcard"
  >
    <div>
      Wildcard pattern
    </div>
  </gl-form-radio-stub>
  <div
    class="gl-ml-6"
  >
    <gl-form-input-stub
      data-testid="webhook_branch_filter_field"
      name="hook[push_events_branch_filter]"
      type="text"
      value=""
    />
  </div>
  <p
    class="custom-control form-text gl-text-subtle"
  >
    <gl-sprintf-stub
      message="Wildcards such as %{WILDCARD_CODE_STABLE} or %{WILDCARD_CODE_PRODUCTION} are supported."
    />
  </p>
  <gl-form-radio-stub
    class="branch-filter-strategy-radio gl-mt-2"
    data-testid="rule_regex"
    value="regex"
  >
    <div>
      Regular expression
    </div>
  </gl-form-radio-stub>
  <div
    class="gl-ml-6"
  />
</gl-form-radio-group-stub>
`;
